mobile/src/helpers/object/group-by-key/group-by-key.helper.ts

```diff
@@ -0,0 +1,24 @@
+const groupByKey = <
+  T extends Record<K, PropertyKey>,
+  K extends keyof {
+    [P in keyof T as T[P] extends PropertyKey ? P : never]: unknown;
+  },
+>(
+  items: T[],
+  key: keyof T,
+): Record<T[keyof T], T[]> => {
+  return items.reduce((hash, item) => {
+    const groupName = item[key];
+
+    if (typeof groupName !== 'string') return hash;
```

Comment:
Don't use if in one line. It becomes hard to read
```suggestion
    if (typeof groupName !== 'string') {
      return hash;
    }
```